### PR TITLE
[DDW-166] Prevent importing more than maximum number of wallets

### PR DESCRIPTION
**Pull request by tomothespian**
This PR prevents importing more than maximum number of wallets

### TODO:
- [x] E2E tests for max wallets import limit

## Screenshots

- Screenshot

Screen Shot 2020-05-11 at 17 41 43

Screen Shot 2020-05-11 at 17 42 02

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1588952809050600)
- [ ] Test if "Max wallets reached" tooltip is displayed correctly. To simulate that you will need to reach max number of wallets possible to import. Create or restore e.g. 28 wallets and then go to import screen and select 2 wallets. So then you will have a total of 30 wallets (2 of them are not restored / imported yet) and then you should not be able to select more wallets for import. In that case when you hover over not selected wallet checkbox you should see "Max wallets reached" tooltip with message.

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by a-rukin** [comment]
@tomothespian can you please increase tooltip width for around 10-20px (maybe a bit more) to avoid one word on a line in english version ?

**Comment by gnpf** [comment]
@tomothespian I was able to import the 31st wallet by staying on the import screen after importing the 30th.
Unfortunately, I couldn't get evidence on the first try, and then Daedalus started to show a blank screen after importing wallets. Not sure if this one is related to this PR, but I'll attach a screenshot of the latter scenario and try to reproduce the previous in another OS to attach the evidence.

wlB5pUbQTC

File: source/renderer/app/i18n/locales/en-US.json [hunk 1]
```diff
@@ -653,6 +653,7 @@
   "wallet.select.import.dialog.importingWallet": "Importing wallet...",
   "wallet.select.import.dialog.linkLabel": "Learn more",
   "wallet.select.import.dialog.linkUrl": "https://iohk.zendesk.com/hc/en-us/articles/900000623463",
+  "wallet.select.import.dialog.maxWalletsReachedTooltip": "Daedalus supports up to 30 wallets. You will need to remove another wallet before you can import this one.",
```
Comment (nikolaglumac) [inline source/renderer/app/i18n/locales/en-US.json:656]: @tomothespian please replace hardcoded "30" limit with variable 🙏

File: source/renderer/app/i18n/locales/ja-JP.json [hunk 2]
```diff
@@ -653,6 +653,7 @@
   "wallet.select.import.dialog.importingWallet": "ウォレットをインポート中…",
   "wallet.select.import.dialog.linkLabel": "もっと知る",
   "wallet.select.import.dialog.linkUrl": "https://iohk.zendesk.com/hc/ja/articles/900000623463",
+  "wallet.select.import.dialog.maxWalletsReachedTooltip": "Daedalusがサポートするウォレット数は最大30です。このウォレットをインポートする前に、いずれかのウォレットを削除してください。",
```
Comment (nikolaglumac) [inline source/renderer/app/i18n/locales/ja-JP.json:656]: @tomothespian please replace hardcoded "30" limit with variable 🙏

File: source/renderer/app/components/wallet/wallet-import/WalletSelectImportDialog.js [hunk 3]
```diff
@@ -145,15 +152,45 @@ export default class WalletSelectImportDialog extends Component<Props> {
...
     if (
       wallet.import.status === WalletImportStatuses.UNSTARTED ||
       wallet.import.status === WalletImportStatuses.PENDING ||
       wallet.import.status === WalletImportStatuses.ERRORED
     ) {
-      const disabled = wallet.name === null || !nameValidator(wallet.name);
+      const invalidWalletName =
+        wallet.name === null || !nameValidator(wallet.name);
+      const walletNotSelectable =
+        isMaxNumberOfWalletsReached &&
```
Comment (nikolaglumac) [inline source/renderer/app/components/wallet/wallet-import/WalletSelectImportDialog.js:170]: @tomothespian here we should check the number of selected wallets and not only if the max number has been reached...

File: source/renderer/app/i18n/locales/en-US.json (hunk 1 above)
Comment (tomothespian) [reply source/renderer/app/i18n/locales/en-US.json]: Done

File: source/renderer/app/i18n/locales/ja-JP.json (hunk 2 above)
Comment (tomothespian) [reply source/renderer/app/i18n/locales/ja-JP.json]: Done

**Comment by tomothespian** [comment]
@a-rukin @gnpf it is ready for another round

**Comment by a-rukin** [comment]
LGTM

**Pull request merged**